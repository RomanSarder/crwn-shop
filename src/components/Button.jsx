import React from 'react'
import styled, { css } from 'styled-components'

var GoogleButtonStyles = css`
    background-color: #4285f4;
    color: white;

    &:hover {
        background: #357ae8;
        border: none;
        color: white;
    }
`

const StyledButton = styled.button`
    min-width: 16.5rem;
    width: auto;
    height: 5rem;
    letter-spacing: 0.5px;
    line-height: 5rem;
    padding: 0 3.5rem 0 3.5rem;
    font-size: 1.5rem;
    background-color: black;
    color: white;
    text-transform: uppercase;
    font-family: 'Open Sans Condensed';
    font-weight: bolder;
    cursor: pointer;
    border: none;

    

    &:hover {
        background-color: white;
        color: black;
        border: 1px solid black;
    }

    ${props => {
        switch (props.appearance) {
            case 'google': {
                return GoogleButtonStyles
            }
            default: return ''
        }
    }}
`

export default function Button({ children, ...otherProps }) {
    return (
        <StyledButton {...otherProps}>
            {children}
        </StyledButton>
    )
}
