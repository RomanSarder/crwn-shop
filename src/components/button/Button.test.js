import '@testing-library/jest-dom'
import 'jest-styled-components'
import { render } from '@testing-library/react'

import Button from './Button'

it('should properly render', () => {
    var { container } = render(
    <Button>
        Test1
    </Button>)

    expect(container.firstChild).toMatchSnapshot()
})

it('should properly render google button', () => {
    var { container } = render(
        <Button appearance="google">
            Test1
        </Button>)
    
        expect(container.firstChild).toMatchSnapshot()
})

it('should properly render inverse button', () => {
    var { container } = render(
    <Button appearance="inverse">
        Test1
    </Button>)

    expect(container.firstChild).toMatchSnapshot()
})