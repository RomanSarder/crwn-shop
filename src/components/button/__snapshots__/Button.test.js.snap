// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render 1`] = `
.c0 {
  min-width: 16.5rem;
  width: auto;
  height: 5rem;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 5rem;
  padding: 0 3.5rem 0 3.5rem;
  font-size: 1.5rem;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  cursor: pointer;
  border: none;
}

.c0:hover {
  background-color: white;
  color: black;
  border: 1px solid black;
}

<button
  class="c0"
>
  Test1
</button>
`;

exports[`should properly render google button 1`] = `
.c0 {
  min-width: 16.5rem;
  width: auto;
  height: 5rem;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 5rem;
  padding: 0 3.5rem 0 3.5rem;
  font-size: 1.5rem;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  cursor: pointer;
  border: none;
  background-color: #4285f4;
  color: white;
}

.c0:hover {
  background-color: white;
  color: black;
  border: 1px solid black;
}

.c0:hover {
  background: #357ae8;
  border: none;
  color: white;
}

<button
  class="c0"
>
  Test1
</button>
`;

exports[`should properly render inverse button 1`] = `
.c0 {
  min-width: 16.5rem;
  width: auto;
  height: 5rem;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 5rem;
  padding: 0 3.5rem 0 3.5rem;
  font-size: 1.5rem;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  cursor: pointer;
  border: none;
  background-color: white;
  color: black;
  border: 2px solid black;
}

.c0:hover {
  background-color: white;
  color: black;
  border: 1px solid black;
}

.c0:hover {
  background-color: black;
  color: white;
}

<button
  class="c0"
>
  Test1
</button>
`;
